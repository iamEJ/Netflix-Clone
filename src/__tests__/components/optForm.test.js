import React from "react";
import { render } from "@testing-library/react";
import OptForm from "../../components/optForm";

describe("<OptForm />", () => {
  it("renders <OptForm /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <OptForm>
        <OptForm.Input placeholder="Enter an email" />
        <OptForm.Button>Try it now</OptForm.Button>
        <OptForm.Text>
          Ready to watch? Enter your email to create or restart your membership.
        </OptForm.Text>
      </OptForm>
    );
    expect(
      getByText(
        "Ready to watch? Enter your email to create or restart your membership."
      )
    ).toBeTruthy();
    expect(getByText("Try it now")).toBeTruthy();
    expect(getByPlaceholderText("Enter an email")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
