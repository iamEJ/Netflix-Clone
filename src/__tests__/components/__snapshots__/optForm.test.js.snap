// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptForm /> renders <OptForm /> with populated data 1`] = `
<div
  class="optForm__Container-sc-1vl0q2b-0 kmAQlr"
>
  <input
    class="optForm__Input-sc-1vl0q2b-1 qCbGJ"
    placeholder="Enter an email"
  />
  <button
    class="optForm__Button-sc-1vl0q2b-2 gpraUo"
  >
    Try it now
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </button>
  <p
    class="optForm__Text-sc-1vl0q2b-3 iIqCCg"
  >
    Ready to watch? Enter your email to create or restart your membership.
  </p>
</div>
`;
